src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs

```diff
@@ -0,0 +1,239 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Resources;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateResourcesCode : Task
+    {
+        [Required]
+        public string ResxFilePath { get; set; }
+
+        [Required]
+        public string IntermediateFilePath { get; set; }
+
+        [Required]
+        public string OutputSourceFilePath { get; set; }
+
+        [Required]
+        public string AssemblyName { get; set; }
+
+        public bool DebugOnly { get; set; }
+
+        public override bool Execute()
+        {
+            bool result = true;
+
+            try
+            {
+                _resxReader = new ResXResourceReader(ResxFilePath);
+                _intermediateFile = IntermediateFilePath + ".temp";
+                _targetStream = File.CreateText(_intermediateFile);
+
+                if (String.Equals(Path.GetExtension(OutputSourceFilePath), ".vb", StringComparison.OrdinalIgnoreCase))
+                {
+                    _targetLanguage = TargetLanguage.VB;
+                }
+
+                _keys = new Dictionary<string, int>();
+                WriteClassHeader();
+                RunOnResFile();
+                WriteDebugCode();
+                WriteClassEnd();
+                Close();
+                ProcessTargetFile();
+            }
+            catch (Exception e)
+            {
+                Close();
+
+                Log.LogMessage(e.Message);
+
+                if (e is System.UnauthorizedAccessException)
+                {
+                    Log.LogMessage("The generated {0} file needs to be updated but the file is read-only. Please checkout this file so it can be updated.", OutputSourceFilePath);
+                }
+                result = false; // fail the task
+            }
+
+            if (result)
+            {
+                // don't fail the task if this operation failed as we just updating intermediate file and the task already did the needed functionality of generating the code
+                try { File.Move(_intermediateFile, IntermediateFilePath); } catch { }
+            }
+
+            return result;
+        }
+        private void Close()
+        {
+            if (_resxReader != null)
+            {
+                _resxReader.Close();
+                _resxReader = null;
+            }
+            if (_targetStream != null)
+            {
+                _targetStream.Close();
+                _targetStream = null;
+            }
+        }
+
+        private void WriteClassHeader()
+        {
+            string commentPrefix = _targetLanguage == TargetLanguage.CSharp ? "// " : "' ";
+            _targetStream.WriteLine(commentPrefix + "This is auto generated file. Please don\u2019t modify manually.");
+            _targetStream.WriteLine(commentPrefix + "The file is generated as part of the build through the ResourceGenerator tool ");
```

Comment:
here is what I have. please update/correct as you think:

"This is auto generated file. Please don't modify manually.
The file is generated as part of the build which takes the project resx resource file
and generated this source code file.
By default the file Resources\\Strings.resx will be used but projects can customize
that by overriding the StringResourcesPath property group.